### DISCUSSION TITLE
Reducing the output from `tar_outdated()`

### DISCUSSION THREAD

**Discussion by TylerGrantSmith** [ropensci/targets#705] [Ideas]
I find myself frequently not caring about all the targets which have become outdated, but only those targets/objects furthest upstream which are outdated.  This would be particularly helpful in large pipelines where an early dependencies produces a large number of outdated nodes.  Could an option be added to `tar_outdated` or perhaps a new function with this kind of functionality?  Maybe I am the only one that has this concern, which is fine too.

In a similar vein, is it possible to identify or filter by the one-or-more causes of a target being outdated?  e.g., did the command change, did the format change, did a dependency get outdated?

**Comment by wlandau** [comment]
[`tar_sitrep()`](https://docs.ropensci.org/targets/reference/tar_sitrep.html) does what I think you are describing.

``` r
library(targets)
tar_script(list(tar_target(x, 1), tar_target(y, x)))
tar_make()
#> • start target x
#> • built target x
#> • start target y
#> • built target y
#> • end pipeline
tar_script(list(tar_target(x, 2), tar_target(y, x))) # invalidate x

tar_sitrep()
#> # A tibble: 2 × 9
#>   name  record always never command depend format iteration file 
#>   <chr> <lgl>  <lgl>  <lgl> <lgl>   <lgl>  <lgl>  <lgl>     <lgl>
#> 1 x     FALSE  FALSE  FALSE TRUE    FALSE  FALSE  FALSE     FALSE
#> 2 y     FALSE  FALSE  FALSE FALSE   FALSE  FALSE  FALSE     FALSE
```

<sup>Created on 2021-11-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
